independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/config/json/RegistriesConfigMapperHelper.java

```diff
@@ -0,0 +1,145 @@
+
+    public static <T> T deserialize(Path p, Class<T> t) throws IOException {
+        if (!Files.exists(p)) {
+            throw new IllegalArgumentException("File " + p + " does not exist");
+        }
+        try (BufferedReader reader = Files.newBufferedReader(p)) {
+            return mapper(p).readValue(reader, t);
+        }
+    }
+
+    public static <T> T deserializeYaml(InputStream is, Class<T> t) throws IOException {
+        return yamlMapper().readValue(is, t);
+    }
+
+    public static <T> T deserializeYaml(Reader reader, Class<T> t) throws IOException {
+        return yamlMapper().readValue(reader, t);
+    }
+
+    public static void main(String[] args) throws Exception {
+        final Path p = serializeConfigSample();
+        //logDeserialized(p);
+        logLoaded(p);
+    }
+
+    private static Path serializeConfigSample() throws IOException {
+        final JsonRegistriesConfig resolver = new JsonRegistriesConfig();
+
+        JsonRegistryConfig qer = new JsonRegistryConfig("qer.acme.com");
+        resolver.addRegistry(qer);
+
+        qer = new JsonRegistryConfig("qer.acme.com");
+        resolver.addRegistry(qer);
+        JsonRegistryMavenConfig mavenConfig = new JsonRegistryMavenConfig();
+        JsonRegistryMavenRepoConfig mvnRepo = new JsonRegistryMavenRepoConfig();
+        mvnRepo.setUrl("https://maven.acme.com");
+        mavenConfig.setRepository(mvnRepo);
+        qer.setMaven(mavenConfig);
+
+        qer = new JsonRegistryConfig("acme.com");
+        resolver.addRegistry(qer);
+        JsonRegistryDescriptorConfig descr = new JsonRegistryDescriptorConfig();
+        qer.setDescriptor(descr);
+        descr.setArtifact(new ArtifactCoords("com.acme", "registry-descriptor", null, "json", "0.1-SNAPSHOT"));
+
+        mavenConfig = new JsonRegistryMavenConfig();
+        mvnRepo = new JsonRegistryMavenRepoConfig();
+        mvnRepo.setUrl("https://acme.com/maven");
+        mavenConfig.setRepository(mvnRepo);
+        qer.setMaven(mavenConfig);
+
+        qer = new JsonRegistryConfig("all");
+        resolver.addRegistry(qer);
+        descr = new JsonRegistryDescriptorConfig();
+        qer.setDescriptor(descr);
+        descr.setArtifact(new ArtifactCoords("org.acme", "registry-descriptor", null, "json", "0.1-SNAPSHOT"));
+
+        mavenConfig = new JsonRegistryMavenConfig();
+        mvnRepo = new JsonRegistryMavenRepoConfig();
+        mvnRepo.setUrl("https://acme.com/rest/api/maven");
+        mavenConfig.setRepository(mvnRepo);
+        qer.setMaven(mavenConfig);
+
+        final Path p = Paths.get(System.getProperty("user.home")).resolve("test-qer-resolver-config.yaml");
+        if (Files.exists(p)) {
+            Files.delete(p);
+        }
+        serialize(resolver, p);
+        return p;
+    }
+
+    private static void logDeserialized(Path p) throws IOException {
+        JsonRegistriesConfig resolver = deserialize(p, JsonRegistriesConfig.class);
+        System.out.println("QER RESOLVER:");
```

Comment:
Remove the `System.out` here (use logger?)

Reply:
It's a left over, removed.